src/molecules/Select/Select.test.tsx

```diff
@@ -1,111 +1,161 @@
 import React from 'react';
-import { mountWithTheme } from '../../test';
-import Select, { Option } from './index';
-const mockClick = jest.fn();
-const mockClickOption = jest.fn();
-const outsideClick = jest.fn();
-
-let citiesData = [
-  { name: 'London', value: 'london' },
-  { name: 'Cardiff', value: 'cardiff' },
-  { name: 'Bristol', value: 'bristol' }
+import { renderWithTheme, fireEvent } from '../../test';
+import Icon from '../../atoms/Icon';
+import Select from './index';
+
+interface Cities {
+  id: number;
+  title: string;
+  country: string;
+  value: string;
+}
+[];
+const cities = [
+  { id: 1, title: 'London', country: 'UK', value: 'london' },
+  { id: 2, title: 'Cardiff', country: 'UK', value: 'cardiff' },
+  { id: 3, title: 'Bristol', country: 'UK', value: 'bristol' },
+  { id: 4, title: 'New York', country: 'US', value: 'newyork' }
 ];
 
-const setup = () =>
-  mountWithTheme(
-    <>
-      <div onClick={outsideClick} id="outside_idiot">
-        I am a idiot
-      </div>
-      <Select
-        handleOptionClick={mockClick}
-        placeholder="I am placeholder"
-        data-qaid="testQaId"
-      >
-        {({ optionClick }: any) =>
-          citiesData.map((cityData, index) => (
-            <Option key={index} value={cityData.value} onClick={optionClick} />
-          ))
-        }
-      </Select>
-    </>
+const setup = (extraProps?: any) =>
+  renderWithTheme(
+    <Select<Cities>
+      data-qaid="select"
+      options={cities}
+      placeholder="Select City"
+      getOptionLabel={o => o.title}
+      {...extraProps}
+    />
   );
 
-describe('Select', () => {
+describe('<Select />', () => {
   it('renders correctly', () => {
-    expect(setup()).toMatchSnapshot();
+    const { queryByTestId } = setup();
+    expect(queryByTestId('select')).toBeInTheDocument();
+    expect(queryByTestId('select-menu')).not.toBeInTheDocument();
+  });
+
+  it('renders correctly with a defaultValue', () => {
+    const { queryByTestId } = setup({
+      defaultValue: cities[2]
+    });
+
+    expect(queryByTestId('select-input')).toHaveValue('Bristol');
+  });
+
+  it('renders correctly with a custom left hand side icon', () => {
+    const { queryByTestId } = setup({
+      renderLeftIcon: <Icon name="helpCircleOutline" />
+    });
+
+    expect(queryByTestId('select-left-icon')).toBeInTheDocument();
+  });
+
+  it('opens the select when clicking the input', () => {
+    const { queryByTestId } = setup();
+
+    fireEvent.click(queryByTestId('select')!);
```

Comment:
I know I've asked before but what does the `!` signify?

Reply:
`fireEvent.click` requires an element to be there - queryByTestId potentially returns null. the `!` signifies that we're confident it'll always be there.